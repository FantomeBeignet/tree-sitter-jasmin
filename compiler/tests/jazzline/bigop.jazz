fn __big(reg u64[4] f) -> reg u64
  requires #[prover=cas] {\sum (ii \in 0:4) (f[ii]) == 3 }
{
 reg u64 test;
 test = f[0] + f[1];
  return test;
}