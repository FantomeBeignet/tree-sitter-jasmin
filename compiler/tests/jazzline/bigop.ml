open Common

let () = parse_and_print "bigop.jazz"  "__big"
