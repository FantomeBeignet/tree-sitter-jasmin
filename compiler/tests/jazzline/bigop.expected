proc __big(test_162@uint64, f_0_150@uint64, f_1_151@uint64, f_2_152@uint64,
           f_3_153@uint64) =
{
 /\[((((((0) + f_0_150) + f_1_151) + f_2_152) + f_3_153) = (3))] && true
}
adds NONE_____173@uint1 test_162@uint64 f_0_150@uint64 f_1_151@uint64;
{
 true && true
}
 